internal/engine/upper.go

```diff
@@ -487,7 +489,17 @@ func handleConfigsReloaded(
 func ensureManifestTargetWithPod(state *store.EngineState, pod *v1.Pod) (*store.ManifestTarget, *store.Pod) {
 	manifestName := model.ManifestName(pod.ObjectMeta.Labels[ManifestNameLabel])
 	if manifestName == "" {
-		return nil, nil
+		// if there's no ManifestNameLabel, then maybe it matches some manifest's ExtraPodLabels
+		for _, m := range state.Manifests() {
+			if m.IsK8s() {
+				for _, lps := range m.K8sTarget().ExtraPodLabels {
+					if lps.AsSelector().Matches(labels.Set(pod.ObjectMeta.GetLabels())) {
```

Comment:
i think the AsSelector() call here points to a bug in the API: ExtraPodLabels should be ExtraPodSelectors? i think?

the k8s API does recognize a distinction between LabelSelectors (https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector) and the LabelSets that they match.

Reply:
The type here is `labels.Selector` rather than `meta.LabelSelector` ([code](https://godoc.org/k8s.io/apimachinery/pkg/labels#Selector)).

Either way, it seems better to make it a selector than a labelset throughout, however:

The `labels.Set` and `labels.Selector` are relatively closely related, and the package has functions to convert [back](https://github.com/windmilleng/tilt/blob/3c4fb73a971b3ea741e1c113865ff5de5130dc90/vendor/k8s.io/apimachinery/pkg/labels/labels.go#L157) and [forth](https://github.com/windmilleng/tilt/blob/3c4fb73a971b3ea741e1c113865ff5de5130dc90/vendor/k8s.io/apimachinery/pkg/labels/labels.go#L61)

k8s exposes an `Equals` methods for `Set`s, but not for `Selector`s, which we need for the diff in the subscriber.

I switched all the `label.Set`s to `label.Selector`s, but then to replace `label.Equals`, I had to implement equality methods, which risk breaking if they add any relevant fields to selectors (or we could use reflection, which might be expensive (or that might not be relevant), and also might break if they add any irrelevant fields to selectors:
```
func setEqual(a, b sets.String) bool {
	if len(a) != len(b) {
		return false
	}

	for k, v := range a {
		if b[k] != v {
			return false
		}
	}

	return true
}

func requirementEqual(a, b labels.Requirement) bool {
	return a.Key() == b.Key() &&
		a.Operator() == b.Operator() &&
		setEqual(a.Values(), b.Values())
}

func requirementsEqual(a, b labels.Requirements) bool {
	if len(a) != len(b) {
		return false
	}

	for i, e1 := range a {
		if !requirementEqual(e1, b[i]) {
			return false
		}
	}

	return true
}
```

We could convert back from selectors to sets for the purpose of equality testing, but that feels kinda weird in the same way that the `AsSelector` here does.

I don't think I have a strong opinion here, and I think I'd lean towards keeping the code as it is in this PR, but I'm fine either way. I just wanted to share the context I had to see if you see a compelling argument one way or another.